### PR TITLE
Update `createObjectsFrom` and `createObjectsFromExternalLayout` to return a list of objects that it creates.

### PR DESCRIPTION
**Pull request by dejw**
This is a small change to the API of `createObjectsFrom` to return a list of objects created, and exposing it further to JS via `createObjectsFromExternalLayout`

This way clients of this API can keep track of the objects that were created from an external layout and manage them further.

Example use-case: being able to unload all objects created from an external layout.

See:
https://forum.gdevelop.io/t/how-can-i-delete-an-external-layout/44511/6?u=dawidfatyga

(FYI: I am located in Zurich so please expect some delay in the response)

### CODE REVIEW

**Comment by Bouh** [comment]
I like this idea, I was thinking first adding the created objects on a layout, but this action doesn't allow to chose where the objects will be created, that's a problem to manage the z-ordering.
But keeping track of created object is a possible solution to my issue.

**Comment by tristanbob** [comment]
I just asked a very similar thing! I want the instances that were created from an external layout to be "picked" so that following actions can apply to the newly created objects.  

Do you think it makes sense to include this feature in this PR?

**Comment by dejw** [comment]
@tristanbob it would be very nice if objects were "picked" if created from an external layout, but this would require changes in Actions related to external layouts, and in my small change I am only changing the JS API, so I think it is out of scope

Looks like there are two issues on this topic already:
- https://github.com/4ian/GDevelop/issues/3572 -- I guess there are some open questions what is the best option here
- https://github.com/4ian/GDevelop/issues/3571 -- this one mentions this is WAI

I have not used "picking" from JS events yet, but I guess there must be a way to pass the resulting list of RuntimeObject into some functions like https://docs.gdevelop.io/GDJS%20Runtime%20Documentation/modules/gdjs.evtTools.object.html#pickAllObjects

@Bouh no code solution would be great, but this would require a more sophisticated changes (like designing a module to track which layout objects came from, somebody mentioned on a bug I cannot find now)

@4ian Hi Florian, As for the review of this PR, what are the next steps? 

Looking at the failed checks it is not clear to me why  "ci/semaphoreci/pr: Fast tests" failed: IDE tests are failing but errors seem unrelated to this change. Tests about types and documentations are passing. Travis CI passes so that is good.

**Comment by 4ian** [comment]
I think it's fine to have this change in JS (as it's not a breaking change in the API). To "pick" objects in future actions/conditions, we would need to have an understanding of all objects that can be potentially used in the external layout. This is indeed more complex and probably out of scope.

cc @D8H

**Review by 4ian — APPROVED** [review]

**Comment by dejw** [comment]
Thanks for a quick review Florian! Looks like I need an approval on one of the workflows that failed. Not sure how to rerun this manually (maybe tests are flaky or failing at head?)

Once done. apparently I will need somebody with "write access" to merge this into the main branch.

**Comment by 4ian** [comment]
Failure seems unrelated, might be because of a mismatch between the webassembly code used in master and in this branch. The rest looks good so I think it's safe to merge! 

Thanks for your contribution :)

**Comment by D8H** [comment]
Usually, making the API easier to use in JS is only a best effort. The issue, in this case, is that it can have an impact on performance of any games that use external layouts. Indeed, it instantiates a new array every time and without pre-allocating it with `length` so the array is probably extended several times. It could trigger garbage collecting and slower games.

I don't know how the array will be used but having every objects in one array doesn't seem very practical.

I think it would be better to rely on variables set on the instances with the editor.

For instance, if the external layout contains a piece of level, there is probably a lot of objects for decoration or collision that don't need to be process and a few enemies that may need some initialization according to a context. These few objects can be tagged with a boolean variable `IsNew` set to `true` in the editor.
If these objects are markers to create enemies depending on the difficulty level, they are destroyed as soon as they are created, so it's doesn't even require to filter them with a variable.

@dejw What is your usecase?

**Comment by 4ian** [comment]
Bad timing here, I just noticed @D8H answer. Sorry for the confusion - I reverted the merge! 

@dejw Let's talk more about your use case to be sure we're not introducing something that would be too risky in terms of performance (or at least, encourage a usage that would not be performant or sub-optimal).

We can either continue here or open another PR to re-introduce the changes.

**Comment by dejw** [comment]
No worries @4ian --

My use-case is quite simple: I am building entire stages inside external layouts, then loading them in JS keeping track of the objects that were created in the process. Then when I want to unload the level I simply iterate over the array and remove the objects from the scene.

This way in one scene I can dynamically load/unload/reload levels.

See: https://forum.gdevelop.io/t/how-can-i-delete-an-external-layout/44511/6?u=dawidfatyga

In other words, my external layouts contain not only a piece, but entire stages, so doing any manual markings with variables is a no-go for me.

If this use-case sounds good to you, how would you suggest changing the code to make sure there is no potential performance regression?

**Comment by D8H** [comment]
A level can be loaded like this:

image

And changed like like this:

image

Does your game need something specific that wouldn't allow this?

**Comment by dejw** [comment]
At this point I don't have any logic that would require to keep 2 levels at the same time and unload them dynamically nor any state that needs to be persisted from level to level (that I would have to impl. on global variables).

So I will use your approach for now as it will work just fine. Thanks for the tip @D8H!

**Comment by 4ian** [comment]
Great tip @D8H, and this probably means that we don't need to extend the interface by returning the objects (unless we find another use case for it later).

I will also add that if at some point you need multiple levels at the same time, you can, instead of using the action "Change the scene", use another action called "Pause and start a new scene" (see https://wiki.gdevelop.io/gdevelop5/all-features/scene/reference/#actions). This will allow to pause a level, run another one on top of it, and then go back to the previous one. 

Thanks both for the good discussion and maybe this can be explained in the wiki at some point or in an article :)
